operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,549 @@
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+pub struct InitializedTemporalRasterAggregation {
+    aggregation_type: AggregationType,
+    window: TimeStep,
+    source: Box<InitializedRasterOperator>,
+    result_descriptor: RasterResultDescriptor,
+    tiling_specification: TilingSpecification,
+}
+
+impl InitializedOperator<RasterResultDescriptor, TypedRasterQueryProcessor>
+    for InitializedTemporalRasterAggregation
+{
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let source_processor = self.source.query_processor()?;
+
+        let res = call_on_generic_raster_processor!(
+            source_processor, p =>
+            TemporalRasterAggregationProcessor::new(
+                self.aggregation_type,
+                self.window,
+                p,
+                self.tiling_specification,
+                self.source.result_descriptor().no_data_value
+            )
+            .boxed().into()
+        );
+
+        Ok(res)
+    }
+}
+
+pub struct TemporalRasterAggregationProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    aggregation_type: AggregationType,
+    window: TimeStep,
+    source: Q,
+    tiling_specification: TilingSpecification,
+    no_data_value: Option<P>,
+}
+
+impl<Q, P> TemporalRasterAggregationProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    fn new(
+        aggregation_type: AggregationType,
+        window: TimeStep,
+        source: Q,
+        tiling_specification: TilingSpecification,
+        no_data_value: Option<f64>,
+    ) -> Self {
+        Self {
+            aggregation_type,
+            window,
+            source,
+            tiling_specification,
+            no_data_value: no_data_value.map(P::from_),
+        }
+    }
+}
+
+impl<Q, P> RasterQueryProcessor for TemporalRasterAggregationProcessor<Q, P>
+where
+    P: Pixel,
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    type RasterType = P;
+
+    fn raster_query<'a>(
+        &'a self,
+        query: crate::engine::QueryRectangle,
+        ctx: &'a dyn crate::engine::QueryContext,
+    ) -> Result<futures::stream::BoxStream<'a, Result<RasterTile2D<Self::RasterType>>>> {
+        match self.aggregation_type {
+            AggregationType::Min => {
+                let spec = TemporalRasterMinAggregationSubQuery {
+                    fold_fn: fold_future::<P, MinCompareFunction>,
+                    no_data_value: self.no_data_value,
+                    initial_value: P::max_value(),
+                    step: self.window,
+                };
+
+                let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                    &self.source,
+                    query,
+                    self.tiling_specification,
+                    ctx,
+                    spec,
+                );
```

Comment:
Could you add the overlap adapter as a method to our [`RasterStreamExt`](https://github.com/geo-engine/geoengine/blob/master/operators/src/adapters/mod.rs)? It wasn't available when this struct was written, but we should add it to improve our stream API. And it will fit nicely here when you can call it directly on `self.source`.

Reply:
Please give me a hint how to do that

Reply:
You could do it similar to [time_multi_fold](https://github.com/geo-engine/geoengine/blob/9cde48cb7c36249e6882653e4c1149cd71dfb293/operators/src/adapters/mod.rs#L36). You know how to call it from the implementation above, so it could work.

Reply:
TemporalRasterAggregationOperator needs a  query processor as input and not a stream so I don't see how I could implement this.